city_scrapers/spiders/alle_port_authority.py

```diff
@@ -0,0 +1,190 @@
+    allowed_domains = ['www.portauthority.org']
+    start_urls = ['https://www.portauthority.org/paac/CompanyInfoProjects/'
+                  'BoardofDirectors/MeetingAgendasResolutions.aspx']
+    event_year = str(datetime.now().year)
+
+    def _build_datatable(self, response):
+        alist_tbody = (response
+                       .xpath('//table[1]/tbody//td')
+                       .extract()
+                       )
+
+        atable = []
+        arow = []
+
+        for item in alist_tbody:
+            tree = html.fragment_fromstring(item)
+            # pdb.set_trace()
+            text = tree.text_content()
+
+            url = tree.xpath('//a/@href')
+            find_att_b = tree.xpath('//b/text()|//strong/text()')
+            if len(find_att_b) >= 1:
+                continue
+            if url:
+                arow.append('{text}: {url}'
+                            .format(text=text, url=url[0]))
+            else:
+                arow.append('{text}'.format(text=unicodedata.normalize("NFKD", text)))
+            if len(arow) == 6:
+                atable.append(arow)
+                arow = []
+        return atable
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` a dict that follows a modified
+        OCD event schema (docs/_docs/05-development.md#event-schema)
+
+        Change the `_parse_id`, `_parse_name`, etc methods to fit your scraping
+        needs.
+        """
+        atable = self._build_datatable(response)
+        for row in atable:
+            data = {
+                'timezone': self.timezone,
+                '_type': 'event',
+                'name': self._parse_name(row),
+                'event_description': self._parse_description(row),
+                'classification': self._parse_classification(row),
+                'start': self._parse_start(row),
+                'end': self._parse_end(row),
+                'all_day': self._parse_all_day(row),
+                'location': self._parse_location(row),
+                'documents': self._parse_documents(row),
+                'sources': self._parse_sources(row),
+            }
+
+            data['status'] = self._generate_status(data)
+            data['id'] = self._generate_id(data)
+
+            yield data
+
+    def _parse_name(self, item):
+        """
+        Parse or generate event name.
+        """
+        return item[0]
+
+    def _parse_description(self, item):
+        """
+        Parse or generate event description.
+        """
+        return ''
+
+    def _parse_classification(self, item):
+        """
+        Differentiate board meetings from public hearings.
+        """
+        meeting_title = item[0].lower()
+        if 'committee' in meeting_title:
+            return COMMITTEE
+        return BOARD
+
+    def _parse_start_datetime(self, item):
+        """
+        Parse the start datetime.
+        """
+        if 'cancel' in item[2].lower():
```

Comment:
From looking at some archives (http://www.portauthority.org/paac/CompanyInfoProjects/BoardofDirectors/2016PublicMeetingArchive.aspx) it seems like they'll leave the time in even if a meeting is canceled sometimes. We should still check and see if the time is available, and use the default meeting time if not

Reply:
My script won't work on any other year except 2018 because the table format keeps changing every or every other year.  I can still check for the time but the date would be missing in year 2018. Would it still be useful???

Reply:
In that case I think it's fine to focus on 2018, thanks for checking!

Reply:
Just to clarify here, this works, but we shouldn't return any meeting that doesn't have a date or time